### PR TITLE
Route53 updater

### PR DESCRIPTION
**Pull request by hugoesthere**
Resolves: https://github.com/Originate/exosphere/issues/869
See: https://github.com/Originate/space-tweet/pull/43 for an example
 
Tested and works:
screen shot 2018-01-04 at 2 15 34 pm

### CODE REVIEW

File: resources/update-route53/main.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	"os/exec"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/route53"
+)
+
+func main() {
+	if len(os.Args) < 3 {
+		panic(errors.New("Not enough arguments. Arguments must be non-empty strings passed into 'route53-updater <service-role> <internal-hosted-zone-name>'"))
```
Comment (charlierudolph) [inline resources/update-route53/main.go]: Lets use `log.Fataln` instead of `panic` as we don't need to have the stack trace printed

**Review by charlierudolph — APPROVED** [review]
Looks great to me. I added in #946 so we can add some documentation for how to use this. Please wait for a review from @alexdavid

File: resources/update-route53/main.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"os"
+	"os/exec"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/route53"
+)
+
+func main() {
+	if len(os.Args) < 3 {
```
Comment (charlierudolph) [inline resources/update-route53/main.go]: How about checking `if len(os.Args) != 3`?

File: terraform/aws/ecs-cluster/iam.tf [hunk 3]
```diff
@@ -43,7 +43,9 @@ resource "aws_iam_role_policy" "ecs_instance" {
         "ecr:GetDownloadUrlForLayer",
         "ecr:BatchGetImage",
         "ecs:StartTask",
-        "autoscaling:*"
+        "autoscaling:*",
+        "route53:ChangeResourceRecordSets",
```
Comment (alexdavid) [inline terraform/aws/ecs-cluster/iam.tf]: I think we should restrict this to only allow changes on the internal hosted zone

File: resources/update-route53/main.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+func main() {
+	if len(os.Args) != 3 {
+		log.Fatalln(errors.New("Wrong number of arguments. Arguments must be non-empty strings passed into 'route53-updater <service-role> <internal-hosted-zone-name>'"))
+	}
+	serviceRole := os.Args[1]
+	internalHostedZoneName := os.Args[2]
+	if serviceRole == "" || internalHostedZoneName == "" {
+		log.Fatalln(errors.New("Service role or internal hosted zone name missing. Both arguments must be non-empty strings passed into 'route53-updater <service-role> <internal-hosted-zone-name>'"))
+	}
+	internalIP, err := exec.Command("curl", "-fsSL", "http://169.254.169.254/latest/meta-data/local-ipv4").Output()
```
Comment (alexdavid) [inline resources/update-route53/main.go]: does it make sense to run this as an external command or just make an http request here?

**Review by alexdavid — APPROVED** [review]

File: terraform/aws/ecs-cluster/iam.tf (hunk 3 above)
Comment (hugoesthere) [reply terraform/aws/ecs-cluster/iam.tf]: How would we restrict it to only the internal hosted zone in aim roles?

**Pull request merged**